### PR TITLE
ENH: Add render functionality

### LINKED ISSUE #1: Initial thoughts
Thoughts on how this package could work.

A simple plugin method:
```python
# plugin/plugin_viz.py
TEMPLATE_DIR = pkg_resources.resource_filename('q2_plugin', 'assets')

def cool_viz(output_dir: str, data: pd.Series) -> None:
    processed_data = foo(data)
    plugin_template = os.path.join(TEMPLATE_DIR, 'index.html')
    outfile = os.path.join(output_dir, 'index.html')
    template_cxt = {'processed_data', processed_data}
    q2design.render(plugin_template, 'plugin_base', outfile, template_cxt)
```

```html+jinja
{# plugin/assets/index.html #}
{% extends base_template %}

{% block title %}The coolest viz!{% endblock title %}
{% block content %}
<h1>A really cool viz</h1>
<ul>
    {% for obj in processed_data %}
    <li>{{ obj.attr_a }} - {{ obj.attr_b }}<li>
    {% endfor %}
</ul>
{% endblock content %}
```

There should be blocks for a plugin dev to provide their own CSS and JS assets. In the call to `q2design.render`, the first arg is the input template, the second arg is the base template to use (provided by `q2design`, the third arg is where to write the resulting file, and the fourth arg is the template context.

### PR DESCRIPTION
**Pull request by jakereps**
~~~Not ready for merge, just staging here.~~~

Resolves #1

### CODE REVIEW

**Comment by jakereps** [comment]
This is ready for a first pass. Want to take it, @thermokarst? I have a working branch ready for `q2-taxa`, that I've been using to test it, once this is good to go as well!

**Comment by jakereps** [comment]
Thanks @thermokarst! Addressed the off-thread review comments on both this and `q2-taxa`!

File: q2templates/_templates.py [hunk 1]
```diff
@@ -5,3 +5,58 @@
...
+    source_files : str, or list of str
+        Files to be rendered and written to the output_dir.
+    output_dir : str
+        The output_dir provided to a visualiation function by the QIIME 2
+        framework.
+    styles : str or list of str, optional
+        The necessary QIIME 2 templates that are used to render the
+        source_files.
+    context : dict, optional
+        The context dictionary to be rendered into the source_files.
```
Comment (thermokarst) [inline q2templates/_templates.py]: Can you add a note here that the same context is passed to all templates to be rendered? Thanks for adding the docstring, super awesome!

File: q2templates/templates/base.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      }
+    </style>
+    {% block head %}
+    {% endblock %}
+  </head>
+  <body>
+    <div class="container-fluid">
+      <div class="row">
+        <div class="col-lg-12">
+          <a href="http://2.qiime.org/">
```
Comment (thermokarst) [inline q2templates/templates/base.html]: Can you replace this with `http://qiime2.org/`?

File: q2templates/templates/base.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            <img class="header" src="./q2templateassets/img/qiime_logo_large.png"
```
Comment (thermokarst) [inline q2templates/templates/base.html]: Can you replace `qiime_logo_large.png` with https://github.com/qiime2/logos/blob/master/raster/qiime2-rect-200.png, and remove the `width` and `height` attrs from the tag?

**Comment by jakereps** [comment]
Thanks @thermokarst, this should be good to go after "testing" 👍 !

**Pull request merged**